### PR TITLE
Comment and styling cleanup

### PR DESCRIPTION
**Pull request by M-Davies**
* Added comments to groovy files
* Ran a markdown linter over our docs
* Ran a spellchecker over both groovy and markdown files

Fixes: #2110 
Signed-off-by: Morgan Davies <morgandavies2020@gmail.com>

### CODE REVIEW

**Review by karianna — APPROVED** [review]
LGTM

**Comment by karianna** [comment]
You've got a merge conflict now

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -190,6 +227,11 @@ class Builder implements Serializable {
         return overrideDocker
     }
 
+    /*
+    Retrieves the dockerImage attribute from the build configurations.
+    This specifies the docker hub org and image to pull or build in case we don't have one stored in this repository.
```
Comment (austin0) [inline pipelines/build/common/build_base_file.groovy:232]: docker hub -> DockerHub

It's styled as `DockerHub` on line 252 so might as well be consistent.

File: pipelines/build/common/build_base_file.groovy [hunk 2]
```diff
@@ -287,10 +341,14 @@ class Builder implements Serializable {
         return configureArgs
     }
 
+    /*
+    Imports the build configurations for the target version based off it's key and variant.
```
Comment (austin0) [inline pipelines/build/common/build_base_file.groovy:345]: It's -> Its

File: pipelines/build/common/config_regeneration.groovy [hunk 3]
```diff
@@ -94,6 +97,11 @@ class Regeneration implements Serializable {
         return configureArgs
     }
 
+    /*
+    Retrieves the dockerImage attribute from the build configurations.
+    This specifies the docker hub org and image to pull or build in case we don't have one stored in this repository.
```
Comment (austin0) [inline pipelines/build/common/config_regeneration.groovy:102]: docker hub -> DockerHub

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 4]
```diff
@@ -598,6 +656,9 @@ class Build {
 
     }
 
+    /*
+    Calculates and writes out the metadata to a file. The metadata defines and summarises a build and the jdk it creates. The adopt v3 api makes use of it in it's endpoints to quickly display information about the jdk binaries that are stored on github.
```
Comment (austin0) [inline pipelines/build/common/openjdk_build_pipeline.groovy:660]: It's -> Its

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 5]
```diff
@@ -712,10 +777,16 @@ class Build {
         return fileName
     }
 
+    /*
+    Executed on a build node, the function checks out the repository and executes the build via ./make-adopt-build-farm.sh
+    Once the build completes, it will calculate it's version output, commit the first metadata writeout and archives the build results.
```
Comment (austin0) [inline pipelines/build/common/openjdk_build_pipeline.groovy:782]: it's -> its

archives -> archive

**Review by austin0 — CHANGES_REQUESTED** [review]
Looks good, thanks for doing this.

File: FAQ.md [hunk 6]
```diff
@@ -58,15 +58,15 @@ In terms of compilers, these are what we currently use for each release:
 
 | Version | OS      | Compiler |
 |---------|---------|----------|
-| JDK8    | Linux   | GCC 4.8 (HotSpot) GCC 7.6 (OpenJ9)                        |
```
Comment (sxa) [inline FAQ.md:61]: I personally think this looks nicer with the final vertical bars lined up (as they are in the rest of the table)

File: FAQ.md [hunk 7]
```diff
@@ -94,13 +94,14 @@ Unless the last release was an LTS one, you will generally want to remove one of
 
 We perform different builds such as the based openjdk (hotspot), builds from the Eclipse OpenJ9 codebase as well as others such as Corretto and SAPMachine. These alternatives are referred to as build variants.
 
-First you will need to add support into the [pipeline files](https://github.com/AdoptOpenJDK/openjdk-build/tree/master/pipelines/build) as well as any environment-specific change syou need to make in the [platform files](https://github.com/AdoptOpenJDK/openjdk-build/tree/master/build-farm/platform-specific-configurations)
```
Comment (sxa) [inline FAQ.md:97]: Thanks for fixing all of my random typos from when I knocked this up initially (I suspect they are mine) - was always intended to be a quick first pass at a FAQ but I'm a bit shocked at how many I put in :-)

File: FAQ.md [hunk 8]
```diff
@@ -94,13 +94,14 @@ Unless the last release was an LTS one, you will generally want to remove one of
...
+First you will need to add support into the [pipeline files](https://github.com/AdoptOpenJDK/openjdk-build/tree/master/pipelines/build) as well as any environment-specific changes you need to make in the [platform files](https://github.com/AdoptOpenJDK/openjdk-build/tree/master/build-farm/platform-specific-configurations)
 For an example, see [this PR where Dragonwell was added](https://github.com/AdoptOpenJDK/openjdk-build/pull/2051/files)
```
Comment (sxa) [inline FAQ.md:98]: Thanks - example PRs for this sort of thing are a great educational tool and I try to include them wherever possible

File: README.md [hunk 9]
```diff
@@ -230,7 +229,7 @@ This is typically used in conjunction with -b.
 
 --use-jep319-certs
```
Comment (sxa) [inline README.md:230]: I think this option has been fully deprecated, but we can deal with that under a separate PR

File: README.md [hunk 10]
```diff
@@ -329,6 +330,7 @@ Alongside the built assets a metadata file will be created with info about the b
     "configure_arguments": <output of bash configure>
```
Comment (sxa) [inline README.md:330]: Not sure I like this being in red - wonder if there's another way we can format it?

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 11]
```diff
@@ -750,15 +830,22 @@ class Build {
         }
     }
 
+    /*
+    Pulls in and applies the activeNodeTimeout parameter.
+    The function will use the jenkins helper nodeIsOnline lib to check every minute if a node with the specified label has come online.
```
Comment (sxa) [inline pipelines/build/common/openjdk_build_pipeline.groovy:835]:
```suggestion
    The function will use the jenkins helper nodeIsOnline lib to check once a minute if a node with the specified label has come online.
```

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 12]
```diff
@@ -824,11 +915,15 @@ class Build {
                                     }
                                     cleanWorkspace = false
                                 }
+
+                                // Use our docker file if we have one
```
Comment (sxa) [inline pipelines/build/common/openjdk_build_pipeline.groovy:919]:
```suggestion
                                // Use our docker file if DOCKER_FILE is defined
```

File: FAQ.md (hunk 6 above)
Comment (sxa) [inline FAQ.md:61]: While it's not a disaster I do personally think this looks easier on the eye lined up, but I wouldn't block it for this change if you disagree :-)

**Review by sxa — CHANGES_REQUESTED** [review]
Lots of nits, several just grammar. Thanks for these changes.

It has reminded me that I need to revisit the docs on makejdk-any-platform.sh ([including jep319](https://github.com/AdoptOpenJDK/openjdk-build/pull/2113/files)) and that I've changed the github team names recently - will update those in a separate PR (After this one has gone in so that I don't cause you any conflicts!)

**Comment by M-Davies** [comment]
Thanks for all the reviews! Looks like I missed quite a bit out 😆

File: README.md (hunk 10 above)
Comment (M-Davies) [reply README.md:330]: I'll stick in a string so it doesn't get parsed as JSON operators
```json
"configure_arguments": "<output of bash configure>"
```

File: pipelines/build/common/build_base_file.groovy (hunk 2 above)
[the remark this answers is not in the document]
Comment (M-Davies) [reply pipelines/build/common/build_base_file.groovy:345]: Dup of https://github.com/AdoptOpenJDK/openjdk-build/pull/2163/files#r511903609

File: pipelines/build/common/config_regeneration.groovy [hunk 13]
```diff
@@ -384,7 +402,7 @@ class Regeneration implements Serializable {
                 // Parse api response to only extract the relevant pipeline
                 getPipelines.jobs.name.each{ pipeline ->
                     if (pipeline.contains("pipeline") && pipeline.contains(versionNumbers[0])) {
-                        // TODO: Paramaterise this
+                        // TODO: Parametrise this
```
[the remark this answers is not in the document]
Comment (M-Davies) [reply pipelines/build/common/config_regeneration.groovy:405]: Strange that my spellchecker picked that up. Oh well, google also says that your suggestion is the right one 🙂

File: FAQ.md (hunk 7 above)
Comment (M-Davies) [reply FAQ.md:97]: To be fair, there weren't that many and I only noticed them when I started running my spellchecker against the docs. It's fairly easy to miss them

File: FAQ.md (hunk 6 above)
Comment (M-Davies) [reply FAQ.md:61]: No I agree. That looked a lot better in my text editor than here. I'll revert this back what it used to be

File: README.md (hunk 9 above)
Comment (M-Davies) [reply README.md:230]: [It looks like we use this a lot](https://github.com/AdoptOpenJDK/openjdk-build/search?q=--use-jep319-certs&type=Code). Does it not have any effect on the builds anymore?

File: pipelines/build/common/build_base_file.groovy [hunk 14]
```diff
@@ -368,7 +443,9 @@ class Builder implements Serializable {
         return true
     }
 
-    // Call job to push artifacts to github
```
[the remark this answers is not in the document]
Comment (M-Davies) [reply pipelines/build/common/build_base_file.groovy:371]: > Purely for consistency on the comments before each function?

Precisely that. More often than not, function descriptions are often multiline so I figured I may as well make them all the same to save time in the future

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 15]
```diff
@@ -210,19 +234,29 @@ class Build {
...
+            // 11.0.9+10-202010192351
             context.println(versionOutput)
 
             return new VersionInfo().parse(versionOutput, buildConfig.ADOPT_BUILD_NUMBER)
         }
         return null
     }
 
+    /*
+    Run the Sign downstream job. We run this job on windows and jdk8 hotspot & jdk13 mac builds.
```
[the remark this answers is not in the document]
Comment (M-Davies) [reply pipelines/build/common/openjdk_build_pipeline.groovy:257]: That's what I parsed the if statement out to. It looks like its true after looking at some various mac builds.

**Review by austin0 — APPROVED** [review]
LGTM

File: README.md (hunk 9 above)
Comment (sxa) [reply README.md:230]: Moving discussion to separate issue https://github.com/AdoptOpenJDK/openjdk-build/issues/2180 - the change was backed out and apparently hasn't been redone yet.

**Review by sxa — APPROVED** [review]
LGTM!

(And I hadn't seen that suggestion that `When a committer comments with "LGTM" they specifically mean “I’ve looked at this patch thoroughly and take as much responsibility as if I wrote it myself”.` before I reviewed this PR)!

Thanks for putting the time into making all these changes - really appreciate it!

**Comment by karianna** [comment]
@M-Davies failing tests on the Groovy...

**Pull request merged**